File: src/main/java/com/imagegame/service/ChallengeCompletionService.java
package com.imagegame.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.imagegame.domain.ChallengeCompletion;
import com.imagegame.dto.ChallengeCompletionDTO;
import com.imagegame.dto.ChallengeCompletionSearchDTO;
import com.imagegame.dto.ChallengeCompletionPageDTO;
import com.imagegame.dto.ChallengeCompletionConvertCriteriaDTO;
import com.imagegame.service.GenericService;
import com.imagegame.dto.common.RequestDTO;
import com.imagegame.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ChallengeCompletionService extends GenericService<ChallengeCompletion, Integer> {

	List<ChallengeCompletion> findAll();

	ResultDTO addChallengeCompletion(ChallengeCompletionDTO challengeCompletionDTO, RequestDTO requestDTO);

	ResultDTO updateChallengeCompletion(ChallengeCompletionDTO challengeCompletionDTO, RequestDTO requestDTO);

    Page<ChallengeCompletion> getAllChallengeCompletions(Pageable pageable);

    Page<ChallengeCompletion> getAllChallengeCompletions(Specification<ChallengeCompletion> spec, Pageable pageable);

	ResponseEntity<ChallengeCompletionPageDTO> getChallengeCompletions(ChallengeCompletionSearchDTO challengeCompletionSearchDTO);
	
	List<ChallengeCompletionDTO> convertChallengeCompletionsToChallengeCompletionDTOs(List<ChallengeCompletion> challengeCompletions, ChallengeCompletionConvertCriteriaDTO convertCriteria);

	ChallengeCompletionDTO getChallengeCompletionDTOById(Integer challengeCompletionId);







}

File: imagegame_ui/src/services/ChallengeCompletionService.js
import http from "../http-common"; 

class ChallengeCompletionService {
  getAllChallengeCompletions(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/challengeCompletion/challengeCompletions`, searchDTO);
  }

  get(challengeCompletionId) {
    return this.getRequest(`/challengeCompletion/${challengeCompletionId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/challengeCompletion?field=${matchData}`, null);
  }

  addChallengeCompletion(data) {
    return http.post("/challengeCompletion/addChallengeCompletion", data);
  }

  update(data) {
  	return http.post("/challengeCompletion/updateChallengeCompletion", data);
  }
  
  uploadImage(data,challengeCompletionId) {
  	return http.postForm("/challengeCompletion/uploadImage/"+challengeCompletionId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ChallengeCompletionService();
